doc/source/programs/gdalwarp.rst

```diff
@@ -493,6 +502,164 @@ The details of how it is taken into account depends on the resampling kernel:
+optimized path in the warper. If you ever see it reporting
+``GDALWarpKernel()::GWKGeneralCase()`` you know things will be relatively slow.
+Basically, the fastest situations are nearest neighbour resampling on 8bit data
+without nodata or alpha masking in effect.
+
+.. caution::
+
+   Increasing the value of the :option:`-wm` param may lead to loss of
+   performance in certain circumstances, particularly when gdalwarping *many*
+   small rasters into a big mosaic. See http://trac.osgeo.org/gdal/ticket/3120
+   for more details
+
+In some cases using ``--config GDAL_CACHEMAX xxx -wm xxx`` can be detrimental.
+A symptom of this is:
+
+::
+
+   ERROR 2: Out of memory allocating 365425784 byte destination buffer.
+
+Our understanding is that 32 bit processes are frequently subject to memory
+fragmentation problems and so even in a process with - in theory - 2GB of heap
+RAM space available it is still often difficult to allocate large blocks of
+memory. In this case with :option:`-wm 500` set the main warp buffers can be
+quite large (364MB for the example) and it seems it has failed to allocate the
+buffer. Use more modest buffers (or not use the options at all) or else get a
+64 bit executable for :program:`gdalwarp`.
+
+Compressed output
+-----------------
+
+In some cases, the output of :program:`gdalwarp` may be much larger than the
+original, even if the same compression algorithm is used. By default,
+:program:`gdalwarp` operates on chunks that are not necessarily aligned with
+the boundaries of the blocks/tiles/strips of the output format, so this might
+cause repeated compression/decompression of partial blocks, leading to lost
+space in the output format.
+
+The situation can be improved by using the ``OPTIMIZE_SIZE`` warping option
+(:option:`-wo OPTIMIZE_SIZE=YES`), but note that depending on the source and
+target projections, it might also significantly slow down the warping process.
+
+Another possibility is to use :program:`gdalwarp` without compression and then
+follow up with :program:`gdal_translate` with compression:
+
+.. code-block:: bash
+
+   gdalwarp infile tempfile.tif ...options...
+   gdal_translate tempfile.tif outfile.tif -co compress=lzw ...etc.
+
+Alternatively, you can use a VRT file as the output format of :program:`gdalwarp`. The
+VRT file is just an XML file that will be created immediately. The
+:program:`gdal_translate` operations will be of course a bit slower as it will do the
+real warping operation.
+
+.. code-block:: bash
+
+   gdalwarp -of VRT infile tempfile.vrt ...options...
+   gdal_translate tempfile.vrt outfile.tif -co compress=lzw ...etc.
+
+This can also be combined this into a single operation by writing the VRT to
+``/vsistdout/`` and piping that to :program:`gdal_translate` and specifying
+``/vsistdin/`` as the input. For example:
+
+.. code-block:: bash
+
+   gdalwarp -q -t_srs EPSG:32611 -of vrt input_file.tif /vsistdout/ | gdal_translate -co compress=lzw /vsistdin/ output_file.tif
+
```

Comment:
clever trick, but I'll remove that. Not totally sure it will work if people want to write tiled output. The alternatives presented before are good enough